File: Lab_3/src/КІ305/Денисяк/Lab3/OfficeCentre.java
/**
 * Lab3 package
 */
package КІ305.Денисяк.Lab3;

/**
 * Class Office Centre implements Building
 */

public class OfficeCentre extends House implements Building{
	private String company_name;
	private int num_office;
	private boolean IsBuildOpen;
	
	/**
	* Constructor
	* @param <code>adress</code> office centre's adress
	* @param <code>floor</code> office centre's floor
	* @param <code>room</code> office centre's room
	* @param <code>door</code> office centres's door
	* @param <code>window</code> office centre's window
	* @param <code>company_name</code> company name of office centre
	* @param <code>num_office</code> number of offices of office centre
	*/
	
	public OfficeCentre(String adress, Floor floor, Room room, Door door, Window window, String company_name, int num_office) {
		super(adress, floor, room, door, window);
		this.company_name = company_name;
		this.num_office = num_office;
		this.IsBuildOpen = false;
	}
	
	/**
	* Constructor
	* @param <code>adress</code> office centre's adress
	* @param <code>num_office</code> number of offices of office centre
	*/
	
	public OfficeCentre(String adress, int num_office) {
		super(new String(adress), new Floor(10, num_office), new Room("Bos's Ccabinet", 20), new Door("Glass"), new Window("Glass"));
		this.company_name = "Mycrosoft";
		this.num_office = num_office;
		this.IsBuildOpen = false;
	}
	
	/**
     * Method open the office center
     */
	
	public void OpenBuild() {
    	if(!this.IsBuildOpen) {
    		this.IsBuildOpen = true;
    		log("Office centre is opened");
    		log(" ");
    	}
    	else {
    		log("Ofiice centre has already opened");
    		log(" ");
    	}
    }
    
	/**
     * Method close the office center
     */
	
	public void CloseBuild() {
		if(this.IsBuildOpen) {
    		this.IsBuildOpen = false;
    		log("Office centre is closed");
    		log(" ");
    	}
    	else {
    		log("Ofiice centre has already closed");
    		log(" ");
    	}
	}
	
	/**
     * Method check information of office centre
     */
	
	public void BuildingInfo() {
		log("Office Centre info:");
		log("Company: " + this.company_name);
		log("Adress: " + GetAdress());
		log("Numer of office " + this.num_office);
		log("Floors: " + GetNumFloor());
		log("Rooms: " + GetNumRoom());
		log("Office centre: " + (IsBuildOpen ? "open" : "close"));
		log(" ");
	}
	
	/**
     * Method perform maintenance on the office centre
     */

	@Override
	public void Perform() {
		// TODO Auto-generated method stub
		log("Performing maintenance on the Office Center");
		log(" ");
	}

	/**
     * Method checking the condition of the Office Center
     */
	
	@Override
	public void CheckCondition() {
		// TODO Auto-generated method stub
		 log("Checking the condition of the Office Center");
		 log(" ");
	}
}

File: Lab_9/OfficeCentre.py
# OfficeCentre.py

from House import House, Floor, Room, Door, Window

# Oficce centre class

class OfficeCentre(House):

    # Constructor

    def __init__(self, address=None, floor=None, room=None, door=None, window=None, company_name="Mycrosoft", num_office=10):
        if address and floor and room and door and window:
            super().__init__(address, floor, room, door, window)
        else:
            super().__init__("Shevchenka 57", Floor(10, num_office), Room("Boss's Cabinet", 20), Door("Glass"), Window("Glass"))
        self.company_name = company_name
        self.num_office = num_office
        self.is_build_open = False

    # Method open building

    def open_build(self):
        if self.is_build_open == False:
            self.is_build_open == True
            print("Office centre is opened\n")
        else:
            print("Office centre has already opened\n")

    # Method close building
    
    def close_build(self):
        if self.is_build_open == True:
            self.is_build_open == False
            print("Office centre is closed\n")
        else:
            print("Office centre has already closed\n")

    # Method show building info

    def building_info(self):
        print("Office centre info:")
        print("Company: %s" %self.company_name)
        print("Adress: %s" %self.get_address())
        print("Number of office: %d" %self.num_office)
        print("Number of floors: %d" %self.get_num_floors())
        print("Number of rooms: %d" %self.get_num_rooms())
        print("Office centre is ", "opened" if self.is_build_open else "closed")

    # Method perform mainstenance

    def perform(self):
        print("Performing maintenance on the Office centre")

    # Method check condirion

    def check_condition(self):
        print("Checking the condition of the Office centre")